contract-deconstruct/src/utils.ts

```diff
@@ -0,0 +1,45 @@
+import { TransactionEvent, ethers, getEthersProvider } from "forta-agent";
+
+var ethProvider: ethers.providers.JsonRpcProvider;
+
+export async function getCreatedContractsTX(txEvent: TransactionEvent){
+
+    const contracts: {
+        deployer: string,
+        contractAddress: string,
+        transaction: string,
+    }[] = [];
+
+    let transaction = txEvent.transaction;
+
+    if(transaction.to === null || transaction.to === "0x0000000000000000000000000000000000000000"){
+        if(!ethProvider) ethProvider = getEthersProvider();
+        const receipt = await ethProvider.getTransactionReceipt(transaction.hash);
+        if(!receipt) return;
+        const contractAddressFromReceipt = receipt.contractAddress;
+        contracts.push({
+            deployer: transaction.from,
+            contractAddress: contractAddressFromReceipt,
+            transaction: transaction.hash,
+        })
+    } else {
+        if(!ethProvider) ethProvider = getEthersProvider();
+        const bytecode = await ethProvider.getCode(transaction.to);
+        if(bytecode.length > 0){
+            for (const trace of txEvent.traces) {
+                if(trace.type === "create"){
```

Comment:
Are create and create2 different ? This is catching only create right ?

Reply:
The trace type detects if it is a call or if it is a contract creation, even if the create or create2 opcode is used, the trace type will still be create.